### PR TITLE
Ed : executables crashe when run with no parameter

### PR DESCRIPTION
**Pull request by nonifier**
Fixes osm2ed when run without parameter

### CODE REVIEW

**Comment by nonifier** [comment]
Currently looking at the other executables that might have the same issue.

File: source/ed/tests/osm2ed_test.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+struct logger_initialized {
+    logger_initialized()   { init_logger(); }
+};
+
+BOOST_GLOBAL_FIXTURE( logger_initialized )
+
+BOOST_AUTO_TEST_CASE(osm2ed_with_no_param_should_start_without_throwing) 
+{
+    const char* argv [] = { "osm2ed_test" };
+    BOOST_CHECK_NO_THROW(ed::connectors::osm2ed(1, argv));
```
Comment (xlqian) [inline source/ed/tests/osm2ed_test.cpp:46]: maybe test the returned value?

File: source/ed/osm2ed_main.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "osm2ed.h"
+
+int main(int argc, const char** argv) {
+    return ed::connectors::osm2ed(argc, argv);
+}
```
Comment (xlqian) [inline source/ed/osm2ed_main.cpp]: newline plz :)

File: source/ed/tests/osm2ed_test.cpp (hunk 1 above)
Comment (nonifier) [reply source/ed/tests/osm2ed_test.cpp:46]: The behavior I intent to fix here, is the throw of a `boost::throw_exception<boost::program_options::required_option>` that triggers a signal abort from `po::notify(vm);`, hence the `BOOST_CHECK_NO_THROW`.

The return value could be tested indeed, but it depends on the command parameters passed in and varies from one another. I'm not sure I want to enforce those behaviors here as it's not relevant to the fix.

File: source/ed/osm2ed_main.cpp (hunk 2 above)
Comment (nonifier) [reply source/ed/osm2ed_main.cpp]: Added in cc7a068320fa0493468adccc2f634394c667a370 :)
But are we not using C++11 feature set ? I thought new lines were not a requirement anymore for end of files.

**Review by kinnou02 — APPROVED** [review]

File: source/ed/osm2ed_main.cpp (hunk 2 above)
Comment (pbougue) [reply source/ed/osm2ed_main.cpp]: We are, but it's more elegant when github doesn't warn :)

**Review by pbougue — COMMENTED** [review]
It's worth having the other exe managed in the same PR

File: source/ed/osm2ed_main.cpp (hunk 2 above)
Comment (nonifier) [reply source/ed/osm2ed_main.cpp]: got it :+1:

**Comment by nonifier** [comment]
Alright! I've justed pushed similar changes for `fare2ed` and `ed2nav` :)
@xlqian & @pbougue : would you mind having a final look at the new changes pliz ?

File: source/ed/tests/osm2ed_test.cpp (hunk 1 above)
Comment (xlqian) [reply source/ed/tests/osm2ed_test.cpp:46]: ok!

**Review by xlqian — APPROVED** [review]

File: source/ed/ed2nav.cpp [hunk 3]
```diff
@@ -273,3 +274,5 @@ int main(int argc, char * argv[])
 
     return 0;
 }
+
+} // namespace ed
```
Comment (pbougue) [inline source/ed/ed2nav.cpp]: minor: newline

File: source/ed/fare2ed.cpp [hunk 4]
```diff
@@ -114,3 +113,6 @@ int main(int argc, char * argv[])
 
     return 0;
 }
+
+} // namespace ed
+
```
Comment (pbougue) [inline source/ed/fare2ed.cpp]: minor: isn't there an extra_line? :)

**Review by pbougue — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

**Pull request merged**